File: base/src/main/java/material/com/base/share/ShareUtils.java
package material.com.base.share;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

/**
 * Created by cangwang on 2017/4/1.
 */

public class ShareUtils {
    public static void shareText(Context ctx,String text){
        Intent sendIntent= new Intent();
        sendIntent.setAction(Intent.ACTION_SEND);
        sendIntent.putExtra(Intent.EXTRA_TEXT,text);
        sendIntent.setType("text/plain");
        ctx.startActivity(Intent.createChooser(sendIntent,"分享至"));
    }

    public static void shareImage(Context ctx,Uri uri){
        Intent sendIntent = new Intent();
        sendIntent.setAction(Intent.ACTION_SEND);
        sendIntent.putExtra(Intent.EXTRA_STREAM,uri);
        sendIntent.setType("image/jpeg");
        ctx.startActivity(Intent.createChooser(sendIntent,"分享至"));
    }
}

File: base/src/main/java/material/com/base/share/ShareUtils.kt
package material.com.base.share

import android.content.Context
import android.content.Intent
import android.net.Uri

/**
 * Created by cangwang on 2017/4/1.
 */

object ShareUtils {
    fun shareText(ctx: Context, text: String) {
        val sendIntent = Intent()
        sendIntent.action = Intent.ACTION_SEND
        sendIntent.putExtra(Intent.EXTRA_TEXT, text)
        sendIntent.type = "text/plain"
        ctx.startActivity(Intent.createChooser(sendIntent, "分享至"))
    }

    fun shareImage(ctx: Context, uri: Uri) {
        val sendIntent = Intent()
        sendIntent.action = Intent.ACTION_SEND
        sendIntent.putExtra(Intent.EXTRA_STREAM, uri)
        sendIntent.type = "image/jpeg"
        ctx.startActivity(Intent.createChooser(sendIntent, "分享至"))
    }
}
